### PR TITLE
The Thingling, Revived

### PR DESCRIPTION
**Pull request by ghost**
:cl: Armhulen coding, Dunham and Onule spriting
add: Changelings have evolved their armblade into a more horrific form to kill.
add: Small changes to headslug > headspider but i can revert if nobody likes this.
del: Which means no more armblade.
/:cl:

[why]: # a lot of people hate changeling for being so strong. changelings SHOULD be strong, but I also feel they should be fair. So that's how I came up with a horror form to replace their armblade so their attacks are like how they are in the thing but also so high costs prevent the changeling from always being able to attack and kill at a moment's notice.

 ## SO IT'S BEEN SOME TIME, WHAT NEEDS TO GET DONE:
new horrors if anyone wants to sprite some (optional!)
refactor transformation for anturke
~balance (basically more health and they slow down a tiny bit if they stay in horror for awhile and i should be okay)~
revert headslug changes if people don't like it
new death sprite hopefully
and that's it!

### CODE REVIEW

**Comment by Buggy123** [comment]
Those commits tho

**Comment by Armhulen** [comment]
it was an arduous task yeah

File: code/game/gamemodes/changeling/powers/horror_form.dm [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/obj/effect/proc_holder/changeling/horror_form //Horror Form: turns the changeling into a terrifying abomination
+	name = "Horror Form"
+	desc = "We tear apart our human disguise, revealing our true form."
+	helptext = "We will become an unstoppable force of destruction. We will turn back into a human after some time."
+	chemical_cost = 75
+	dna_cost = 3
+	req_human = 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/changeling/powers/horror_form.dm]: TRUE

File: code/game/gamemodes/changeling/powers/horror_form.dm [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+/obj/effect/proc_holder/changeling/horror_form/sting_action(mob/living/carbon/human/user)
+	if(!user || user.notransform)
+		return 0
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/changeling/powers/horror_form.dm]: FALSE

File: code/game/gamemodes/changeling/powers/horror_form.dm [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+						"<span class='userdanger'>We cast off our petty shell and enter our true form!</span>")
+	var/mob/living/simple_animal/hostile/true_changeling/new_mob = new(get_turf(user))
+	var/datum/antagonist/changeling/ling_datum = user.mind.has_antag_datum(/datum/antagonist/changeling)
+	new_mob.real_name = ling_datum.changelingID
+	new_mob.name = new_mob.real_name
+	new_mob.stored_changeling = user
+	user.loc = new_mob
+	user.status_flags |= GODMODE
+	user.mind.transfer_to(new_mob)
+	return 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/changeling/powers/horror_form.dm]: TRUE

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+	speak_emote = list("says with one of its faces")
+	emote_hear = list("says with one of its faces")
+	icon = 'icons/mob/changeling.dmi'
+	icon_state = "horror1"
+	icon_living = "horror1"
+	icon_dead = "horror_dead"
+	speed = -0.5
+	gender = NEUTER
+	a_intent = "harm"
+	stop_automated_movement = 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: TRUE

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+	status_flags = 0
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: should be inherited, no?

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+	..()
+	if(statpanel("Status"))
+		if(stored_changeling)
+			var/time_left = TRUE_CHANGELING_FORCED_REFORM - time_spent_as_true
+			time_left = Clamp(time_left, 0, INFINITY)
+			stat(null, "Time Remaining: [time_left]")
+		stat(null, "Ignoring Gravity: [wallcrawl ? "YES" : "NO"]")
+
+/mob/living/simple_animal/hostile/true_changeling/death()
+	..(1)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: TRUE?

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+				for(var/mob/M in view(7, src))
+					flash_color(M, flash_color = list("#db0000", "#db0000", "#db0000", rgb(0,0,0)), flash_time = 5)
+				new /obj/effect/gibspawner/human(get_turf(src))
+				revive() //Changelings can self-revive, and true changelings are no exception
+
+/mob/living/simple_animal/hostile/true_changeling/mob_negates_gravity()
+	return wallcrawl
+
+/mob/living/simple_animal/hostile/true_changeling/adjustFireLoss(amount)
+	playsound(src, 'sound/creatures/ling_scream.ogg', 100, 1)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: spam-prevention please.

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+		to_chat(M, "<span class='warning'>We are not able to change back at will!</span>")
+		return FALSE
+	M.visible_message("<span class='warning'>[M] suddenly crunches and twists into a smaller form!</span>", \
+					"<span class='danger'>We return to our human form.</span>")
+	M.stored_changeling.forceMove(get_turf(M))
+	M.mind.transfer_to(M.stored_changeling)
+	M.stored_changeling.Unconscious(200)
+	M.stored_changeling.status_flags &= ~GODMODE
+	qdel(M)
+	return 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: TRUE

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+						"<span class='userdanger'>We lack the power to maintain this form! We helplessly turn back into a human...</span>")
+		stored_changeling.loc = get_turf(src)
+		mind.transfer_to(stored_changeling)
+		stored_changeling.Unconscious(300) //Make them helpless for some time
+		stored_changeling.status_flags &= ~GODMODE
+		qdel(src)
+	else
+		visible_message("<span class='warning'>[src] lets out a waning scream as it falls, twitching, to the floor.</span>", \
+						"<span class='userdanger'>We have fallen! We begin the revival process... (this will take around 45 seconds.)</span>")
+		spawn(450)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: ❗️

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 5 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: is not, i don't think

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: i can test!

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 9 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: before i didn't know how to addtimers but i do now so i'll fix this tonight

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 7 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: there's cooldowns on attacking and guns, it should be wailing a lot if people are

AND BEFORE ANYONE ASKS, I DID ADD IF(STAT)

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 5 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: you're correct, simple animals have CANPUSH

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 7 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: what about being on fire?

**Comment by Ratvar** [comment]
Make horrorform leave blood traces automatically? More work for detective, more spook
Make transformation revert happen per bodypart and be slow? Run around with thicc legs if you are unlucky and have armblade if it's your day for 30 seconds.

**Comment by RemieRichards** [comment]
@Armhulen I don't want Horror3 used, please remove it.

**Comment by Armhulen** [comment]
Sure

**Comment by Armhulen** [comment]
Sucks that was a personal favorite

**Comment by Spaceinaba** [comment]
2 is the coolest tbh

**Comment by ExcessiveUseOfCobblestone** [comment]
1 is the classic option

**Comment by JJRcop** [comment]
In reality commit count is not something to worry about

**Comment by Qbopper** [comment]
\>not immediately autoclosed due to GBP

this isn't an arm PR

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 7 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: Simplemobs can't be on fire but it didn't screech in space so

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: > ling
> can't be on fire

KILL IT

serious note, +1 then

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 5 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: fistbump me bro

**Comment by RemieRichards** [comment]
*cough* sprite *cough*

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 6 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: unsure

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 7 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: more burn DAMAGE

**Comment by Armhulen** [comment]
@RemieRichards these things



they take time

**Comment by PKPenguin321** [comment]
i still think armblade should be left out of this, it fills a niche and isn't even what i'd call overpowered

**Comment by Armhulen** [comment]
the niche being what thingling accomplishes but at 40 chems? if you're talking doors i can add a changeling ability for opening doors

**Comment by PKPenguin321** [comment]
maybe incorporate opening doors to horror form.

i guess if i dont even use armblade it's kind of silly for me to be defending it like this but you gotta see how spending every chem you have to transform into another being completely is different from just having an on-demand OK weapon.

**Comment by Armhulen** [comment]
it's a pretty strong weapon, i wouldn't call something that equates to an energy sword and OK weapon. still, changelings should be about using deception to get away and not an offensive ability. that's why i want it to be something you commit to when you have someone alone, to kill them. having a good weapon like that makes the changeling a lot less about evasion than it should be when most of the time you can just kill anyone onto you

**Comment by PKPenguin321** [comment]
i guess you're right. i do enjoy evasion ling more than combat ling. i figured this just made combat ling more obtuse but if the goal is to make ling have a violent getaway ability then i like it.

**Comment by Armhulen** [comment]
loading tgstation.dme
loading interface\skin.dmf
loading map_files\generic\CentCom.dmm
loading map_files\generic\SpaceStation.dmm
loading map_files\generic\Space.dmm
loading map_files\generic\SpaceDock.dmm
loading map_files\Mining\Lavaland.dmm
loading map_files\generic\City_of_Cogs.dmm
saving tgstation.dmb (DEBUG mode)
tgstation.dmb - 0 errors, 0 warnings (1/22/18 11:56 pm)

travis failed??

**Comment by JJRcop** [comment]
Travis isn't building the right commit

**Comment by Armhulen** [comment]
so ignore it?

**Comment by JJRcop** [comment]
A maintainer should try to trigger it again, you can also do it yourself with `git commit --allow-empty -m "Travis rebuild"`

**Review by ayyourboibilly — COMMENTED** [review]
gud

**Comment by scrubs2009** [comment]
Right now there's really no reason to "gib th succ" to more than one or two people because all it gets you is rerolls. What if each crew member succ'd gave you an extra DNA point to spend?

**Comment by davethwave** [comment]
I've just thought of a great new way to gib lings in case of emergency! 100/100 potassium water grenade have a gib radius of 1 but you would probably want to make a room just for gibbing them or have easy access to a space suit for it.

**Comment by Armhulen** [comment]
Both of those posts have NOTHING to do with horror form (and how the fuck is having a grenade on you at all times AND BUILDING A SPECIFIC CHANGELING GIB CENTER a great new way???)

**Comment by Nabski89** [comment]
@scrubs2009 Succing everyone you kill is optimal play. You may not like it, but this is what extra TC looks like.

**Comment by Armhulen** [comment]
literally just ignore travis it's retarded by the way

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+	time_spent_as_true++ //Used for re-forming
+	if(stored_changeling && time_spent_as_true >= TRUE_CHANGELING_FORCED_REFORM)
+		death() //After a while, the ling'll revert back without being able to control it
+
+/mob/living/simple_animal/hostile/true_changeling/Stat()
+	..()
+	if(statpanel("Status"))
+		if(stored_changeling)
+			var/time_left = TRUE_CHANGELING_FORCED_REFORM - time_spent_as_true
+			time_left = Clamp(time_left, 0, INFINITY)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: CLAMP

**Comment by ExcessiveUseOfCobblestone** [comment]
poor travis. Works so hard to help ur pr, tells you what you did wrong, and you call him retarded :cry:

File: code/modules/mob/living/simple_animal/hostile/true_changeling.dm (hunk 10 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/true_changeling.dm]: WHY WOULD WE DO THAT

**Comment by Armhulen** [comment]
can you delete that comment please i can't take this bad press

**Comment by JJRcop** [comment]
Aww how can travis help if you don't ask him? Please forgive him for playing hookie earlier, he had a cold. 😰

**Comment by RemieRichards** [comment]
@Armhulen I hate to be pushy, but with all these commits you keep doing, and none doing what I've asked, I can't help but feel you're trying to sneak it through.

**Comment by Armhulen** [comment]
@RemieRichards you asked me to remove horror2, and i did?

**Comment by RemieRichards** [comment]
3, and IconDiffBot didn't update so if you have done it, I've not seen so.

**Comment by Armhulen** [comment]
right 3, icondiffbot does not update to the sprite changes, the commit i removed it was...

image
this one!

**Comment by RemieRichards** [comment]
the last commit I see is TRUE x 3, must just be github then.

**Comment by Armhulen** [comment]
and also just between you and me i really love horror3 so i hope you come around to it eventually! it's like a uuh scorpion or something but it feels like classic horror form ling but newer

aaaaaanyways

**Comment by davethwave** [comment]
I've been in a few rounds with this test merged but I never learned about the different forms. What varieties are there in horror form and how are they picked?

**Comment by Armhulen** [comment]
random form every time you transform into a horror. also i fixed the speed and added some more flair so it's coming back

**Comment by Armhulen** [comment]
sorry i haven't worked on this in a tiny bit i'm having deep conniptions about where i want to take thinglings and how i should balance it

**Comment by Armhulen** [comment]
i'll get an idea after xhuis does something cool with shadowlings and come back that's my plan

**Comment by Buggy123** [comment]
You look at your armhulen and sigh

**Comment by PKPenguin321** [comment]
Yall mind if I steal that bg_changeling.dmi

**Pull request closed without merging**